trieste/models/gpflux/models.py

```diff
@@ -127,7 +125,19 @@ def scheduler(epoch: int, lr: float) -> float:
             }
 
         self._model_gpflux = model
-
+        # inputs and targets need to be redone with a float64 dtype to avoid setting the keras
+        # backend to float64, this is likely to be fixed in GPflux, see issue:
+        # https://github.com/secondmind-labs/GPflux/issues/76
+        self._model_gpflux.inputs = tf.keras.Input(
```

Comment:
Is this sufficient, or should you copy the other parameters too (batch_size, sparse, ragged, etc)?

Reply:
as far as I can see it should be fine,  this issue would typically surface when calling a method or trying to train the model - here training works, calling all the methods we have works fine in the tests - can you think of a test that would fail?

Reply:
I guess when inputs are correctly set, dtype is propagated to other layers

Reply:
Looks like gpflux doesn't set any of them. I guess if we're planning to fix this in gpflux anyway then this is ok for now.